pandas/_libs/tslibs/dtypes.pyx

```diff
@@ -278,6 +278,19 @@ class NpyDatetimeUnit(Enum):
     NPY_FR_GENERIC = NPY_DATETIMEUNIT.NPY_FR_GENERIC
 
 
+cpdef NPY_DATETIMEUNIT get_supported_reso(NPY_DATETIMEUNIT reso):
+    # If we have an unsupported reso, return the nearest supported reso.
+    if reso == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
+        # TODO: or raise ValueError? trying this gives unraisable errors, but
```

Comment:
Yeah I would support raising. I image this would cause silent failures. Okay to follow up

Reply:
An alternative is to just return `NPY_FR_GENERIC` and have the caller check for it? Or I imagine other code paths would raise (albeit probably with not an entirely clear message)